test/framework/autoscaler_helpers.go

```diff
@@ -0,0 +1,276 @@
+		WatchDeploymentLogsByName(ctx, WatchDeploymentLogsByNameInput{
+			GetLister:  input.WorkloadClusterProxy.GetClient(),
+			ClientSet:  input.WorkloadClusterProxy.GetClientSet(),
+			Deployment: deployment,
+			LogPath:    filepath.Join(input.ArtifactFolder, "clusters", input.WorkloadClusterProxy.GetName(), "logs", deployment.GetNamespace()),
+		})
+	}
+}
+
+// AddScaleUpDeploymentAndWaitInput is the input for AddScaleUpDeploymentAndWait.
+type AddScaleUpDeploymentAndWaitInput struct {
+	ClusterProxy ClusterProxy
+}
+
+// AddScaleUpDeploymentAndWait create a deployment that will trigger the autoscaler to scale up and create a new machine.
+func AddScaleUpDeploymentAndWait(ctx context.Context, input AddScaleUpDeploymentAndWaitInput, intervals ...interface{}) {
+	By("Create a scale up deployment requiring 2G memory for each replica")
+
+	// gets the node size
+	nodes := &corev1.NodeList{}
+	workers := 0
+	Expect(input.ClusterProxy.GetClient().List(ctx, nodes)).To(Succeed(), "failed to list nodes")
+	var memory *resource.Quantity
+	for _, n := range nodes.Items {
+		if _, ok := n.Labels[nodeRoleControlPlane]; ok {
+			continue
+		}
+		memory = n.Status.Capacity.Memory()
+		workers++
+	}
+	Expect(workers).To(Equal(1), "AddScaleUpDeploymentAndWait assumes only one worker node exist")
+	Expect(memory).ToNot(BeNil(), "failed to get memory for the worker node")
+
+	// creates a deployment requesting more memory than the worker has, thus triggering autoscaling
+	var replicas int64 = 5
+	podMemory := resource.NewQuantity(memory.Value()/(replicas-1), resource.BinarySI)
+
+	scalelUpDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "scale-up",
+			Namespace: metav1.NamespaceDefault,
+			Labels: map[string]string{
+				"app": "scale-up",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32(int32(replicas)),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "scale-up",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "scale-up",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "busybox",
+							Image: "busybox",
+							Resources: corev1.ResourceRequirements{
+								Requests: map[corev1.ResourceName]resource.Quantity{
+									corev1.ResourceMemory: *podMemory,
+								},
+							},
+							Command: []string{"/bin/sh", "-c", "echo \"up\" & sleep infinity"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	By("Scale up deployment created")
+	Expect(input.ClusterProxy.GetClient().Create(ctx, scalelUpDeployment)).To(Succeed(), "failed to create the scale up pod")
+
+	By("Wait for the scale up deployment to become ready (this implies machines to be created)")
+	WaitForDeploymentsAvailable(ctx, WaitForDeploymentsAvailableInput{
+		Getter:     input.ClusterProxy.GetClient(),
+		Deployment: scalelUpDeployment,
+	}, intervals...)
```

Comment:
could also delete and wait for the MachineDeployment to scale down fwiw https://github.com/openshift/hypershift/blob/main/test/e2e/autoscaling_test.go#L22.